src/modules/RootPart.rb

```diff
@@ -483,11 +482,16 @@ def RunFSCKonJFS(mount_type, device, error_message)
     # @return [String] nil on success, error description on fail
     def MountPartition(mount_point, device, mount_type, fsopts = "")
       if mount_type == ""
-        # storage-ng
-        # FIXME: knowing the fstab string for the filesystem type should be
-        # responsibility of the upcoming Y2Storage::FilesystemType (to be
-        # added as part of the "yast-storage-ng as a libstorage wrapper" change)
-        mount_type = probed.partitions.with(name: device).filesystems.first.to_s
+
+        # FIXME
+        #
+        # Note that the code below will get passed unmodified fstab entries
+        # (like 'UUID=2f61fdb9-f82a-4052-8610-1eb090b82098') for device names
+        # and typically not match anything due to this.
+        #
+        # See also the comment in #update_staging_filesystem! below.
+        #
+        mount_type = fstype_for_device(probed, device) || ""
```

Comment:
> Are we sure that Filesystems::Type#to_s is exactly the string to use in fstab?

Yes. In the same way you'd expect the symbol for 'ext2' to be `:ext2`. What else should it be than the canonical filesystem name used by the kernel everywhere? it's just the natural thing to expect.

You could set up an alias but that will just irritate people. At least I would want to know what the difference between `mount_string` and `to_s` is.